src/grid/tests/test_onedgrid.py

```diff
@@ -1,61 +1,71 @@
-"""Onedgrid tests file."""
-from unittest import TestCase
+# -*- coding: utf-8 -*-
+# GRID is a numerical integration library for quantum chemistry.
+#
+# Copyright (C) 2011-2017 The GRID Development Team
+#
+# This file is part of GRID.
+#
+# GRID is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 3
+# of the License, or (at your option) any later version.
+#
+# GRID is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, see <http://www.gnu.org/licenses/>
+#
+# --
+"""Tests one-dimensional grids."""
 
-from grid.onedgrid import (
-    GaussChebyshev,
-    GaussLaguerre,
-    GaussLegendre,
-    HortonLinear,
-    generate_onedgrid,
-)
+
+from grid.onedgrid import GaussChebyshev, GaussLaguerre, GaussLegendre, HortonLinear
 
 import numpy as np
-from numpy.testing import assert_allclose
+
+import pytest
 
 from scipy.special import roots_legendre
 
 
-class TestOneDGrid(TestCase):
-    """OneDGrid test class."""
-
-    def setUp(self):
-        """Test setup function."""
-        ...
-
-    def test_generate_ondgrid(self):
-        """Place holder tests for generate_onedgrid."""
-        generate_onedgrid(0)
-
-    def test_gausslaguerre(self):
-        """Test Guass Laguerre polynomial grid."""
-        points, weights = np.polynomial.laguerre.laggauss(10)
-        weights = weights * np.exp(points) * np.power(points, 0)
-        grid = GaussLaguerre(10)
-        assert_allclose(grid.points, points)
-        assert_allclose(grid.weights, weights)
-
-    def test_gausslengendre(self):
-        """Test Guass Lengendre polynomial grid."""
-        points, weights = roots_legendre(10)
-        grid = GaussLegendre(10)
-        assert_allclose(grid.points, points)
-        assert_allclose(grid.weights, weights)
-
-    def test_gausschebyshev(self):
-        """Test Guass Chebyshev polynomial grid."""
-        points, weights = np.polynomial.chebyshev.chebgauss(10)
-        weights = weights * np.sqrt(1 - np.power(points, 2))
-        grid = GaussChebyshev(10)
-        assert_allclose(grid.points, np.sort(points))
-        assert_allclose(grid.weights, weights)
-
-    def test_horton_linear(self):
-        """Test horton linear grids."""
-        grid = HortonLinear(10)
-        assert_allclose(grid.points, np.arange(10))
-        assert_allclose(grid.weights, np.ones(10))
-
-    def test_errors_raise(self):
-        """Test errors raise."""
-        with self.assertRaises(ValueError):
-            GaussLaguerre(10, -1)
+def test_gauss_laguerre():
```

Comment:
I'd like it to be a unittest class more. It's a buildint test class with some nice feature like `mock` etc, which we probably can use in the future. It also keeps the tests file style consistent.

Reply:
Let's talk about this later. I think it is important to be consistent across modules (iodata, gbasis, and chemtools all use the test functions), and unless we are using the extra features of the test classes that you were referring to, I am not sure we should bother with them (as they look/seem/are more complicated and the extra indentation is a waste of line characters). Also, some of those features might be achievable with pytest. That's just my opinion, but I am not an expert. We should wait until we talk with @tovrstra and also see what @kimt33 has to say.

Reply:
I believe the `unittest.TestCase` class has little added value when using pytest. When running tests with `python -m unittest <modulename>`, I believe the classes are mandatory and one should also be using its methods like `self.assertTrue(...)`, etc. The main goal of frameworks like pytest (and previously nosetests) is to make unit testing easier, e.g. without `TestCase` classes and by encouraging simple usage of the `assert` statement instead of the large number of `TestCase.assert*` methods.

Also the pytest fixtures are easier to use than the `setUp` and `tearDown` methods of the `TestCase` class. See for example the following test in iodata, which needs a temporary directory: https://github.com/theochem/iodata/blob/master/iodata/test/test_cube.py#L99
Just by adding `tmpdir` argument, the setup and teardown of the temporary directory are taken care of. With the old unittest module, this would require one to write `setUp` and `tearDown` methods. I've also used pytest fixtuyres in tinydft to parameterize tests, which worked very well. 

The `unittest.mock` module can be used also without the `TestCase` class, see e.g. examples here: https://docs.python.org/dev/library/unittest.mock.html#quick-guide.  The pytest-mock plugin is designed to make mock easier to use with pytest, see https://pypi.org/project/pytest-mock/

If there are other features of the `TestCase` class, please let me know. As far as I can tell, it should be easier to write unit tests without these classes, without losing features of the old `unittest` module.

Reply:
@tovrstra Thank you very much for the explanation. I do agree `pytest` work like a charm for testing and I like it from the very beginning.
`Unittest` are `pytest` are not exclusive to each other. We can combine both to get all the edgy features if we need. 
So here's my defence for not to change this test file.

1. `unittest` is a buildin module. If the feathers needed are offered in `unittest`, there's little motivation for me resorting to a 3rd party module. (though I put `pytest` in setup.py, it's not really a dependence).

2. For `Grid`, tests are frame free. One can use `pytest`, `nosetests`, or even 'unittest' to run all the tests without problem. User/developer are free to use their preferred test framework.

3. To support `unittest` style test, it only need one extra line. And to me, it doesn't look terribly ugly or complicated.